service/src/main/java/uk/gov/hmcts/reform/fpl/service/GeneratedOrderService.java

```diff
@@ -71,24 +99,67 @@ public GeneratedOrder addDocumentToOrder(GeneratedOrder order, Document document
     }
 
     public Map<String, Object> getOrderTemplateData(CaseData caseData) {
-        return ImmutableMap.<String, Object>builder()
+        ImmutableMap.Builder<String, Object> orderTemplateBuilder = new ImmutableMap.Builder<>();
+
+        //Scalable for future order types
+        switch (caseData.getOrderTypeAndDocument().getOrderType()) {
+            case BLANK_ORDER:
+                orderTemplateBuilder
+                    .put("orderType", BLANK_ORDER)
+                    .put("orderTitle", defaultIfNull(caseData.getOrder().getOrderTitle(), "Order"))
+                    .put("childrenAct", "Section 31 Children Act 1989")
+                    .put("orderDetails", caseData.getOrder().getOrderDetails());
+                break;
+            case CARE_ORDER:
+                orderTemplateBuilder
+                    .put("orderType", CARE_ORDER)
+                    .put("orderTitle", "Care Order")
+                    .put("childrenAct", "Children Act 1989")
+                    .put("orderDetails", careOrderDetails(getChildrenDetails(caseData).size(),
+                        caseData.getCaseLocalAuthority()));
+                break;
+            default:
+        }
+        orderTemplateBuilder
             .put("familyManCaseNumber", caseData.getFamilyManCaseNumber())
             .put("courtName", getCourtName(caseData.getCaseLocalAuthority()))
-            .put("orderTitle", defaultIfNull(caseData.getOrder().getOrderTitle(), "Order"))
-            .put("orderDetails", caseData.getOrder().getOrderDetails())
             .put("todaysDate", dateFormatterService.formatLocalDateTimeBaseUsingFormat(time.now(), "d MMMM yyyy"))
             .put("judgeTitleAndName", JudgeAndLegalAdvisorHelper.formatJudgeTitleAndName(
                 caseData.getJudgeAndLegalAdvisor()))
             .put("legalAdvisorName", JudgeAndLegalAdvisorHelper.getLegalAdvisorName(
                 caseData.getJudgeAndLegalAdvisor()))
             .put("children", getChildrenDetails(caseData))
             .build();
+
+        return orderTemplateBuilder.build();
+    }
+
+    public String generateDocumentFileName(String type) {
+        return type.toLowerCase().replaceAll("[()]", "").replaceAll("[ ]", "_") + ".pdf";
+    }
+
+    public String mostRecentUploadedOrderDocumentUrl(final List<Element<GeneratedOrder>> orders) {
+        return getLast(orders.stream()
+            .filter(Objects::nonNull)
+            .map(Element::getValue)
+            .filter(Objects::nonNull)
+            .collect(toList()))
+            .getDocument().getBinaryUrl();
     }
 
     private String getCourtName(String courtName) {
         return hmctsCourtLookupConfiguration.getCourt(courtName).getName();
     }
 
+    private String getLocalAuthorityName(String caseLocalAuthority) {
+        return localAuthorityNameLookupConfiguration.getLocalAuthorityName(caseLocalAuthority);
+    }
+
+    private String careOrderDetails(int numOfChildren, String caseLocalAuthority) {
```

Comment:
Confusing as we have orderDetails in the Order type, consider renaming.

Reply:
Well this is actually sort of intended. This is the details for a care order, it's just fixed, so on the template it will always show this. Whereas for a C21 order the orderDetails are not fixed and the user can enter them.